pacman_mirrors/pacman_mirrors.py

```diff
@@ -165,64 +156,129 @@ def command_line_parse(self):
         if args.branch:
             self.config["branch"] = args.branch
 
-        if args.mirror_dir:
-            self.config["mirror_dir"] = args.mirror_dir
-            self.default_mirror_dir = self.config["mirror_dir"]
-
-        self.available_countries = sorted(
-            os.listdir(self.config["mirror_dir"]))
+        if args.timeout:
+            self.max_wait_time = args.timeout
 
-        if args.geoip:
-            self.geolocation = True
+        if args.quiet:
+            self.quiet = True
 
-        if args.country:
-            country = args.country.split(",")
-            if country == ["Custom"]:
-                self.config["only_country"] = country
-            elif country == ["all"]:
-                self.config["only_country"] = []
-            else:
-                try:
-                    self.validate_country_list(country,
-                                               self.available_countries)
-                    self.config["only_country"] = country
-                except argparse.ArgumentTypeError as err:
-                    parser.error(err)
+        if args.mirror_dir:
+            self.config["mirror_dir"] = args.mirror_dir
 
         if args.output:
             if args.output[0] == "/":
                 self.config["mirror_list"] = args.output
             else:
                 self.config["mirror_list"] = os.getcwd() + "/" + args.output
-            self.default_mirror_list = self.config["mirror_list"]
 
         if args.interactive:
             self.interactive = True
             if not os.environ.get("DISPLAY") or not GTK_AVAILABLE:
                 self.no_display = True
+        # geoip and country are mutually exclusive
+        if args.geoip:
+            self.geolocation = True
 
-        if args.timeout:
-            self.max_wait_time = args.timeout
+        if args.country and not args.geoip:
+            self.config["only_country"] = args.country.split(",")
 
```

Comment:
You're right, they are mutually exclusive, but I think a better approach will be to exit and inform the user they can't use --country and --geoip at the same time.